scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,442 @@
+                                       create_dir=True)
+
+    # Merge all echos path into a list
+    maps = [args.in_altnp, args.in_altpn, args.in_mtoff, args.in_negative,
+            args.in_positive, args.in_t1w]
+
+    maps_flat = (args.in_altnp + args.in_altpn + args.in_mtoff +
+                 args.in_negative + args.in_positive + args.in_t1w)
+
+    jsons = [curr_map.replace('.nii.gz', '.json')
+             for curr_map in maps_flat]
+
+    # check data
+    assert_inputs_exist(parser, jsons + maps_flat)
+    for curr_map in maps[1:]:
+        if len(curr_map) != len(maps[0]):
+            parser.error('Not the same number of echoes per contrast')
+
+    # Set TR and FlipAngle parameters for ihMT (positive contrast)
+    # and T1w images
+    parameters = [set_acq_parameters(maps[4][0].replace('.nii.gz', '.json')),
+                  set_acq_parameters(maps[5][0].replace('.nii.gz', '.json'))]
+
+    # Fix issue from the presence of NaN into array
+    np.seterr(divide='ignore', invalid='ignore')
+
+    # Define reference image for saving maps
+    ref_img = nib.load(maps[4][0])
+
+    # Define contrasts maps names
+    contrasts_name = ['altnp', 'altpn', 'reference', 'negative', 'positive',
+                      'T1w']
+    if args.filtering:
+        contrasts_name = [curr_name + '_filter'
+                          for curr_name in contrasts_name]
+
+    # Compute contrasts maps
+    computed_contrasts = []
+    for idx, curr_map in enumerate(maps):
+        computed_contrasts.append(compute_contrasts_maps(
+                                  curr_map, filtering=args.filtering))
+
+        nib.save(nib.Nifti1Image(computed_contrasts[idx].astype(np.float64),
+                                 ref_img.affine, ref_img.header),
+                 os.path.join(args.out_dir, 'Contrasts_ihMT_maps',
+                              args.id_subj + '_' + contrasts_name[idx]
+                              + '.nii.gz'))
+
+    # Compute and thresold ihMT maps
+    ihMTR, ihMTsat = compute_ihMT_maps(computed_contrasts, parameters)
+    ihMTR = threshold_ihMT_maps(ihMTR, computed_contrasts, args.in_mask,
+                                0, 100, [4, 3, 1, 0, 2])
+    ihMTsat = threshold_ihMT_maps(ihMTsat, computed_contrasts, args.in_mask,
+                                  0, 10, [4, 3, 1, 0])
+
+    # Compute and thresold non-ihMT maps
+    MTR, MTsat = compute_MT_maps(computed_contrasts, parameters)
+    for curr_map in MTR, MTsat:
+        curr_map = threshold_ihMT_maps(curr_map, computed_contrasts,
+                                       args.in_mask, 0, 100, [4, 2])
+
+    # Save ihMT and MT images
+    img_name = ['ihMTR', 'ihMTsat', 'MTR', 'MTsat']
+
+    if args.filtering:
+        contrasts_name = [curr_name + '_filter'
+                          for curr_name in contrasts_name]
+
+    img_data = ihMTR, ihMTsat, MTR, MTsat
+    for img_to_save, name in zip(img_data, img_name):
+        nib.save(nib.Nifti1Image(img_to_save.astype(np.float64),
```

Comment:
Do we really need np.float64 ? Unless the value are known to be below 1x10^6 and lower we go for np.float32